### PR TITLE
[chore] add goleak tests for prometheusreceiver

### PR DESCRIPTION
**Pull request by iblancasa**
Related to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30438

### CODE REVIEW

File: receiver/prometheusreceiver/package_test.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	"testing"
+
+	"go.uber.org/goleak"
+)
+
+// The IgnoreTopFunction call prevents catching the leak generated by opencensus
+// defaultWorker.Start which at this time is part of the package's init call.
+// See https://github.com/census-instrumentation/opencensus-go/issues/1191 for more information.
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m, goleak.IgnoreTopFunction("go.opencensus.io/stats/view.(*worker).start"))
```
[the remark this answers is not in the document]
Comment (iblancasa) [reply receiver/prometheusreceiver/package_test.go]: Actually is not needed. Thanks!

[the remark this answers is not in the document]
Comment (iblancasa) [reply receiver/prometheusreceiver/package_test.go]: I was not able to reproduce it locally but yes, it seems it is needed:
```sh
=== FAIL: .  (0.00s)
PASS
goleak: Errors on successful test run: found unexpected goroutines:
[Goroutine 10 in state select, with go.opencensus.io/stats/view.(*worker).start on top of the stack:
go.opencensus.io/stats/view.(*worker).start(0xc00023ad80)
	/home/runner/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:292 +0x1a5
created by go.opencensus.io/stats/view.init.0 in goroutine 1
	/home/runner/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:34 +0xfd
]
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver	247.888s
testing: warning: no tests to run
PASS
goleak: Errors on successful test run: found unexpected goroutines:
[Goroutine 6 in state select, with go.opencensus.io/stats/view.(*worker).start on top of the stack:
go.opencensus.io/stats/view.(*worker).start(0xc000606e00)
	/home/runner/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:[292](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8201802738/job/22431325011?pr=31630#step:8:293) +0x1a5
created by go.opencensus.io/stats/view.init.0 in goroutine 1
	/home/runner/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:34 +0xfd
]
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver	0.518s

DONE 2 runs, 553 tests, 68 skipped, 1 failure in 323.546s
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
```

**Review by crobert-1 — APPROVED** [review]
Thanks for adding, @iblancasa!

**Review by atoulme — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

**Comment by crobert-1** [comment]
_Lint failures were unrelated, they should be fixed by rebasing._

**Review by codeboten — APPROVED** [review]

**Pull request merged**